docs/spec/parallel_dask.rst

```diff
@@ -0,0 +1,60 @@
+.. _parallel_dask:
+
+============================
+Parallel Execution with Dask
+============================
+
+
+Queries
+-------
+
+Kartothek executes queries with a task for every parquet file to read. These
+files are filtered by the predicate's restrictions on partitions and indexed
+columns.
+
+.. image:: /images/kartothek_read_dispatch.png
+  :width: 600
+  :alt: Kartothek dispatches tasks by parquet file
+
+
+For every file its range of values (stated in the parquet footer) is matched
+against the query. (See :doc:`/spec/predicate_pushdown`)
+
+.. image:: /images/kartothek_read_pushdown.png
+  :width: 600
+  :alt: Kartothek pushes down parts of the query to the parquet layer
+
+If a file's range of values don't match, the task returns an empty Dataframe.
+Otherwise the files are loaded and their data filtered.
+
+.. image:: /images/kartothek_read_final.png
+  :width: 600
+  :alt: Kartothek filters the resulting dataframes in every task
+
+These dataframes then make up the result.
+
+
+
+Writes
+------
+
+If ``shuffle == False`` (default) the partitioning of the dataset between the
+tasks doesn't change, the graph of tasks stays the same as it was for creating
+the dataset that was supplied to be stored. Every partition of data of a task
+has to be distributed across the different folders for the partitioned columns.
+So every task writes multiple distinct files, one in every folder with the data
+that has the specific of values in the partitioned columns which belong into
+this folder.
+
+In the other case, if ``shuffle == True``, the data has to be regrouped
+according to the ``partition_on``, ``bucket_by`` and ``bucket_num`` parameters.
+This is handled by Dask and, in the case of running on a cluster, involves
+sending the data over network between the workers. More information can be
+found in `Dask's documentation <https://docs.dask.org/en/latest/dataframe-groupby.html>`_.
+
+At the end of either case the results of the writes are collected and finally
+atomically commited to Kartothek's dataset.
+
+.. image:: /images/kartothek_partition_on.png
+  :width: 600
+  :alt: Kartothek's storing process
```

Comment:
I don't think it's clear for a new users what the pictures represent. Please explain what's going on in the pictures (e.g. this row group is ignored because predicates don't match...)